app/scripts/modules/core/src/pipeline/config/triggers/pubsub/pubsub.trigger.ts

```diff
@@ -1,19 +1,35 @@
 import { IController, module } from 'angular';
 
 import { PIPELINE_CONFIG_PROVIDER, PipelineConfigProvider } from 'core/pipeline/config/pipelineConfigProvider';
-import { IPubsubTrigger } from '@spinnaker/core';
+import {
+  PUBSUB_SUBSCRIPTION_SERVICE,
+  PubsubSubscriptionService,
+} from 'core/pubsub';
+import {
+  IPubsubTrigger,
+} from 'core/domain';
 
 class PubsubTriggerController implements IController {
   public pubsubSystems = ['kafka', 'google'];
+  public pubsubSubscriptions: string[];
+  public subscriptionsLoaded = false;
 
-  constructor(public trigger: IPubsubTrigger) {
+  constructor(public trigger: IPubsubTrigger,
+              pubsubSubscriptionService: PubsubSubscriptionService) {
     'ngInject';
+
+    this.subscriptionsLoaded = false;
+    pubsubSubscriptionService.getPubsubSubscriptions().then(subscriptions => {
+      this.subscriptionsLoaded = true;
+      this.pubsubSubscriptions = subscriptions;
+    }).catch(() => this.pubsubSubscriptions = []);
```

Comment:
Should flip `subscriptionsLoaded` even in the case where the call failed (the spinner should stop spinning and you should see an empty dropdown).

You can use `.finally` here:

```
pubsubSubscriptionService.getPubsubSubscriptions()
  .then(subscriptions => this.pubsubSubscriptions = subscriptions)
  .catch(() => this.subscriptions = [])
  .finally(() => this.subscriptionsLoaded = true);
```